src/rdkafka_telemetry_encode.c

```diff
@@ -183,105 +259,180 @@ void *rd_kafka_telemetry_encode_metrics(rd_kafka_t *rk, size_t *size) {
+        instrumentationScope.name.arg             = (void *)rd_kafka_name(rk);
         instrumentationScope.version.funcs.encode = &encode_string;
-        instrumentationScope.version.arg          = &"Example scope version";
+        instrumentationScope.version.arg = (void *)rd_kafka_version_str();
 
         scopeMetrics.has_scope = true;
         scopeMetrics.scope     = instrumentationScope;
 
-        opentelemetry_proto_metrics_v1_Metric metric =
-            opentelemetry_proto_metrics_v1_Metric_init_zero;
-
-        opentelemetry_proto_metrics_v1_Sum sum =
-            opentelemetry_proto_metrics_v1_Sum_init_zero;
-
-        opentelemetry_proto_metrics_v1_NumberDataPoint data_point =
-            opentelemetry_proto_metrics_v1_NumberDataPoint_init_zero;
-        data_point.which_value =
-            opentelemetry_proto_metrics_v1_NumberDataPoint_as_int_tag;
-        data_point.value.as_int   = calculate_connection_creation_total(rk);
-        data_point.time_unix_nano = rd_clock();
-
-        //    TODO: Add data point attributes as needed
-        //    opentelemetry_proto_common_v1_KeyValue attribute =
-        //    opentelemetry_proto_common_v1_KeyValue_init_zero;
-        //    attribute.key.funcs.encode = &encode_string;
-        //    attribute.key.arg = "type";
-        //    attribute.has_value = true;
-        //    attribute.value.which_value =
-        //    opentelemetry_proto_common_v1_AnyValue_string_value_tag;
-        //    attribute.value.value.string_value.funcs.encode = &encode_string;
-        //    attribute.value.value.string_value.arg = "heap";
-        //
-        //    data_point.attributes.funcs.encode = &encode_key_value;
-        //    data_point.attributes.arg = &attribute;
-
-        sum.data_points.funcs.encode = &encode_number_data_point;
-        sum.data_points.arg          = &data_point;
-        sum.aggregation_temporality =
-            opentelemetry_proto_metrics_v1_AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE;
-        sum.is_monotonic = true;
-
-        metric.which_data = opentelemetry_proto_metrics_v1_Metric_sum_tag;
-        metric.data.sum   = sum;
-
-        metric.description.funcs.encode = &encode_string;
-        metric.description.arg          = metric_description;
-
-        metric.name.funcs.encode = &encode_string;
-        metric.name.arg          = metric_name;
+        metrics = rd_malloc(sizeof(opentelemetry_proto_metrics_v1_Metric *) *
+                            metrics_to_encode_count);
+        data_points =
+            rd_malloc(sizeof(opentelemetry_proto_metrics_v1_NumberDataPoint *) *
+                      metrics_to_encode_count);
+        metric_names = rd_malloc(sizeof(char *) * metrics_to_encode_count);
+        size_t i;
+
+        for (i = 0; i < metrics_to_encode_count; i++) {
+                metrics[i] =
+                    rd_malloc(sizeof(opentelemetry_proto_metrics_v1_Metric));
+                memset(metrics[i], 0,
```

Comment:
Use rd_calloc above instead of malloc and memset

Reply:
Updated